### PR TITLE
[MLIR][Python] add f8E5M2 and tests for np_to_memref

### PR DESCRIPTION
**Pull request by PhrygianGates**
add f8E5M2 and tests for np_to_memref

### CODE REVIEW

File: mlir/python/mlir/runtime/np_to_memref.py [hunk 1]
```diff
@@ -65,6 +72,11 @@ def to_numpy(array):
     ), f"bfloat16 requires the ml_dtypes package, please run:\n\npip install ml_dtypes\n"
     if array.dtype == BF16:
         return array.view("bfloat16")
+    assert not (
+        array.dtype == F8E5M2 and ml_dtypes is None
+    ), f"float8_e5m2 requires the ml_dtypes package, please run:\n\npip install ml_dtypes\n"
```
Comment (makslevental) [inline mlir/python/mlir/runtime/np_to_memref.py:77]: nit: you don't need the `f` here

**Review by makslevental — APPROVED** [review]

File: mlir/python/mlir/runtime/np_to_memref.py (hunk 1 above)
Comment (makslevental) [reply mlir/python/mlir/runtime/np_to_memref.py:77]: Well then you can remove in both

**Comment by PhrygianGates** [comment]
Screenshot 2024-08-27 at 10 05 33 AM

Hi experts! Since I'm not familiar with the LLVM PR merge workflow, does this mean I still need approvals to merge?

**Comment by makslevental** [comment]
> Hi experts! Since I'm not familiar with the LLVM PR merge workflow, does this mean I still need approvals to merge?

It just meant someone had to approve a CI job for you. I've done it.

**Pull request merged**